drivers: usb: usb_dc_nrfx: add usbd worker thread name.

markCwatson opened the pull request:
Note: there remains other drivers using worker threads that do not explicitly name them. May create a different issue to capture that work since this issue pertains to the usb_dc_nrfx specifically.

Signed-off-by: Mark Watson <markus.c.watson@gmail.com>

Fixes: #43330

markCwatson committed:
drivers: usb: usb_dc_nrfx: add usbd worker thread name.

I added a thread name to the usbd worker thread.

Signed-off-by: Mark Watson <mwatson@prosaris.ca>

Fixes: #43330

desowin commented on drivers/usb/device/usb_dc_nrfx.c:
```diff
@@ -1918,6 +1918,7 @@ static int usb_init(const struct device *arg)
 			   K_KERNEL_STACK_SIZEOF(usbd_work_queue_stack),
 			   CONFIG_SYSTEM_WORKQUEUE_PRIORITY, NULL);
 
+	(void)k_thread_name_set(&usbd_work_queue.thread, "usbd_workq");
```
k_thread_name_set() is called without casting the result in other places, why would the cast be needed here?

markCwatson replied:
See 'drivers\modem\gsm_ppp.c' as an example where it's casted. Can easily remove the cast.

markCwatson replied:
Removed the cast

carlescufi replied:
Please remove the `(void)`, I see most callers don't use it.

desowin reviewed (commented):
Please rebase the commits. Currently this PR contains merge commit and that conflicts with Zephyr development model. Otherwise, it is fine, thanks.

jfischer-no reviewed (changes requested):
Please read https://docs.zephyrproject.org/latest/contribute/guidelines.html
Use 'git log' to get familiar with commit style used in the project and subsystem.
Look at the merged PR for USB subsystem to get familiar how others format PR title.

markCwatson commented:
> Please rebase the commits. Currently this PR contains merge commit and that conflicts with Zephyr development model. Otherwise, it is fine, thanks.

Uuuummm.... I tried rebasing the commit according to the [guideline](https://docs.zephyrproject.org/latest/contribute/guidelines.html), but I'm not sure if I did it right, and now there are a bunch of reviewers added to this PR LOL

jfischer-no commented:
> > Please rebase the commits. Currently this PR contains merge commit and that conflicts with Zephyr development model. Otherwise, it is fine, thanks.
> 
> Uuuummm.... I tried rebasing the commits according to the [guideline](https://docs.zephyrproject.org/latest/contribute/guidelines.html#:~:text=If%20reviewers%20do%20request%20changes%20to%20your%20patch%2C%20you%20can%20interactively%20rebase%20commit(s)%20to%20fix%20review%20issues.%20In%20your%20development%20repo%3A), but I don't think I did it right because now there are a bunch of reviewers added to this PR LOL Sorry about that everyone

That does not hurt anyone.
Using `git rebase -i ....` you should only see your commit in the rebase editor.  Use `git push --force origin add_name_to_usbd_work_queue` after rebase.

markCwatson commented:
@jfischer-no @desowin I think I cleaned this PR up to where it's supposed to be. Let me know if not. Thanks for your patience.

jfischer-no reviewed (changes requested):
Please, use 'git log' to get familiar with commit style used in the project and subsystem.

markCwatson commented:
> Please, use 'git log' to get familiar with commit style used in the project and subsystem.

I got some assistance on Discord, and I think I the commit message contents are acceptable now. Thanks.

markCwatson commented:
@jfischer-no How do things look now?

jfischer-no reviewed (dismissed):
> @jfischer-no How do things look now?

It looks acceptable to me, thank you.

markCwatson commented:
@desowin Are you able to give the second approval?

carlescufi reviewed (changes requested):
Please remove the `(void)`

henrikbrixandersen commented:
> Please remove the `(void)`

Why? Having `(void)` makes it explicit, that the return value is ignored. Similar to `ARG_UNUSED()`.

markCwatson commented:
I went ahead and removed the explicit `(void)` since more people suggested I make this change than not.

carlescufi commented:
> > Please remove the `(void)`
> 
> Why? Having `(void)` makes it explicit, that the return value is ignored. Similar to `ARG_UNUSED()`.

Just for consistency with the rest of the tree, where most calls don't use it.

desowin reviewed (changes requested):
```
drivers: usb: usb_dc_nrfx: removed explicit cast of return value.

I also removed the cast for this function in drivers\modem\gsm_ppp.c
```

This is not really how it should be done. If you remove the case in `gsm_ppp.c`, then that should be a fully separate commit with its own title and message. Also introducing a commit with a line that is then changed in subsequent commit in the same PR without a solid reason is not that great (the original commit should have been amended to remove the `(void)`).

jfischer-no commented:
> > > Please remove the `(void)`
> > 
> > 
> > Why? Having `(void)` makes it explicit, that the return value is ignored. Similar to `ARG_UNUSED()`.
> 
> Just for consistency with the rest of the tree, where most calls don't use it.

There is no consistency in this regard. This has something to do with the belief in MISRA...

markCwatson commented:
> ```
> drivers: usb: usb_dc_nrfx: removed explicit cast of return value.
> 
> I also removed the cast for this function in drivers\modem\gsm_ppp.c
> ```
> 
> This is not really how it should be done. If you remove the case in `gsm_ppp.c`, then that should be a fully separate commit with its own title and message. Also introducing a commit with a line that is then changed in subsequent commit in the same PR without a solid reason is not that great (the original commit should have been amended to remove the `(void)`).

OK no problem. Such strict requirements on clean commit history are new to me. I know the contribution guideline discusses how to edit a single commit, but I was confused when following the procedure. Using interactive rebasing with git is also new to me. All good though: happy and eager to learn, so I'll see if I can remove the extra commit and modify the original with the cast removed. Thanks.

markCwatson commented:
twister is failing. Not sure why. Wasn't before rebasing with origin/main. I can't imagine why my change would cause this. Also, I am developing on Windows, and I cannot run twister on my machine.

carlescufi commented:
> > ```
> > drivers: usb: usb_dc_nrfx: removed explicit cast of return value.
> > 
> > I also removed the cast for this function in drivers\modem\gsm_ppp.c
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This is not really how it should be done. If you remove the case in `gsm_ppp.c`, then that should be a fully separate commit with its own title and message. Also introducing a commit with a line that is then changed in subsequent commit in the same PR without a solid reason is not that great (the original commit should have been amended to remove the `(void)`).
> 
> @desowin OK no problem. Such strict requirements on clean commit history are new to me. I know the contribution guideline discusses how to edit a single commit, but I was confused when following the procedure. Using interactive rebasing with git is also new to me. All good though: happy and eager to learn, so I'll see if I can remove the extra commit and modify the original with the cast removed. Thanks.

It's OK, we can keep this PR simple, it's your first one. Feel free to send a second one later.
